anchore_engine/services/catalog/catalog_impl.py

```diff
@@ -2177,6 +2199,37 @@ def add_or_update_image(
     return ret
 
 
+def is_new_tag(image_record: dict, registry: str, repo: str, tag: str):
+    """
+    Returns false if the provided registry, repo, tag tuple exactly match those fields in one of the image_detail objects in
+    image_record. If no such match is possible, return true.
+    """
+
+    # If we are missing an image_details block entirely or it is None, assume this is a new tag
+    if "image_detail" not in image_record or not image_record["image_detail"]:
```

Comment:
(nit) slightly optimized check for image_record and image_detail in a single condition
```
if image_record and image_record.get("image_detail")
```

Reply:
I've made that change